src/layout/MasterLayout.cpp

```diff
@@ -1071,42 +1071,71 @@ std::any CHyprMasterLayout::layoutMessage(SLayoutMessageHeader header, std::stri
         recalculateMonitor(header.pWindow->m_iMonitorID);
 
     } else if (command == "orientationnext") {
-        const auto PWINDOW = header.pWindow;
-
-        if (!PWINDOW)
-            return 0;
+        runOrientationCycle(&header, nullptr, 1);
+    } else if (command == "orientationprev") {
+        runOrientationCycle(&header, nullptr, -1);
+    } else if (command == "orientationcycle") {
+        runOrientationCycle(&header, &vars, 1);
+    }
 
-        prepareLoseFocus(PWINDOW);
+    return 0;
+}
 
-        const auto PWORKSPACEDATA = getMasterWorkspaceData(PWINDOW->m_iWorkspaceID);
+// If vars is null, we use the default list
+void CHyprMasterLayout::runOrientationCycle(SLayoutMessageHeader* header, CVarList* vars, int direction) {
```

Comment:
ditto

Reply:
Done

Reply:
Here I left CVarList as a pointer because I need to allow null to be passed for orientationnext and orientationprev:
```
std::vector<eOrientation> cycle;
if (vars != nullptr)
    buildOrientationCycleVectorFromVars(cycle, *vars);
```

What I could actually do is to remove the new command (orientationcycle) and just pass my new list to orientationext and orientationprev. That would remove the need to pass CVarList as null.

Like this:
```
bind = $mainMod, J, layoutmsg, orientationnext left right center
bind = $mainMod, K, layoutmsg, orientationprev
```
**Current configs would keep working.**